src/Controls/src/Core/Shell/ShellSection.cs

```diff
@@ -410,16 +410,27 @@ async Task PrepareCurrentStackForBeingReplaced(ShellNavigationRequest request, S
 								continue;
 						}
 
-						// This is the page that we will eventually get to once we've finished
-						// modifying the existing navigation stack
-						// So we want to fire appearing on it						
-						navPage.SendAppearing();
-
+						// We use this inside ModalNavigationManager to indicate that we're going to be popping multiple
+						// modal pages so we don't want it to fire any appearing events
```

Comment:
Could we do something in the UI from the UITest, like update a Label Text, to validate the correct behavior with the Appearing method?

Reply:
This existing test covers the modal scenario here
- PoppingModalStackFiresAppearingOnRevealedModalPage 

I've added
- PoppingModalStackFiresAppearingOnRevealedNonModalPage

To cover popping multiple pages down to a shell page to ensure appearing still fires